contracts/protocol/facets/DisputeResolverHandlerFacet.sol

```diff
@@ -188,40 +193,194 @@ contract DisputeResolverHandlerFacet is IBosonAccountEvents, ProtocolBase {
+        }
+
+        if (updateApplied) {
+            // Notify watchers of state change
+            emit DisputeResolverUpdateApplied(
+                _disputeResolver.id,
+                disputeResolver,
+                disputeResolverPendingUpdate,
+                sender
+            );
+        }
+    }
+
+    /**
+     * @notice Opt-in to a pending dispute resolver update
+     *
+     * Emits a DisputeResolverUpdateApplied event if successful.
+     *
+     * Reverts if:
+     * - The dispute resolver region of protocol is paused
+     * - Addresses are not unique to this dispute resolver
+     * - Caller address is not pending update for the field being updated
+     * - No pending update exists for this dispute resolver
+     *
+     * @param _disputeResolverId - disputeResolver id
+     * @param _fieldsToUpdate - fields to update, see DisputeResolverUpdateFields enum
+     */
+    function optInToDisputeResolverUpdate(
+        uint256 _disputeResolverId,
+        DisputeResolverUpdateFields[] calldata _fieldsToUpdate
+    ) external disputeResolversNotPaused nonReentrant {
+        // Cache protocol lookups and sender for reference
+        ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
+        address sender = msgSender();
+
+        // Get disputeResolver pending update
+        (bool exists, DisputeResolver storage disputeResolverPendingUpdate) = fetchDisputeResolverPendingUpdate(
+            _disputeResolverId
+        );
+
+        require(exists, NO_PENDING_UPDATE_FOR_ACCOUNT);
+
+        bool updateApplied;
+
+        // Get storage location for disputeResolver
+        (, DisputeResolver storage disputeResolver, ) = fetchDisputeResolver(_disputeResolverId);
+
+        for (uint256 i = 0; i < _fieldsToUpdate.length; i++) {
+            DisputeResolverUpdateFields role = _fieldsToUpdate[i];
+
+            // Approve admin update
+            if (role == DisputeResolverUpdateFields.Admin && disputeResolverPendingUpdate.admin != address(0)) {
+                require(disputeResolverPendingUpdate.admin == sender, UNAUTHORIZED_CALLER_UPDATE);
+
+                preUpdateDisputeResolverCheck(_disputeResolverId, sender, lookups);
+
+                // Delete old disputeResolver id by admin mapping
+                delete lookups.disputeResolverIdByAdmin[disputeResolver.admin];
+
+                // Update admin
+                disputeResolver.admin = sender;
+
+                // Store new disputeResolver id by admin mapping
+                lookups.disputeResolverIdByAdmin[sender] = _disputeResolverId;
+
+                // Delete pending update admin
+                delete disputeResolverPendingUpdate.admin;
+
+                updateApplied = true;
+            }
+
+            // Approve operator update
+            if (role == DisputeResolverUpdateFields.Operator && disputeResolverPendingUpdate.operator != address(0)) {
```

Comment:
This should be `else if` since in each iteration `role` will match only one of possible fields.

Same is true in `optInToSellerUpdate`

Reply:
nice catch, done